src/delay.rs

```diff
@@ -40,89 +42,54 @@ impl Delay {
     pub fn reset(&mut self, deadline: Instant) {
         self.deadline = deadline;
         self.initialized = false;
-        if let Some(task) = &self.task {
-            task.notify()
-        }
     }
 }
 
 impl Future for Delay {
-    type Item = ();
-    type Error = IoError;
+    type Output = Result<(), IoError>;
 
-    fn poll(&mut self) -> Result<Async<Self::Item>, Self::Error> {
+    fn poll(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
         if !self.initialized {
             let now = Instant::now();
             let duration = if self.deadline > now {
                 self.deadline - now
             } else {
-                return Ok(Async::Ready(()));
+                return Poll::Ready(Ok(()));
             };
             self.timerfd
                 .set_state(TimerState::Oneshot(duration), SetTimeFlags::Default);
             self.initialized = true;
         }
-        try_ready!(self.timerfd.poll_read());
-        Ok(Async::Ready(()))
+        let mut buf = [0u8; 8];
+        let mut buf = ReadBuf::new(&mut buf);
+        ready!(Pin::new(&mut self.as_mut().timerfd).poll_read(cx, &mut buf)?);
+        Poll::Ready(Ok(()))
```

Comment:
Is the intention here to ignore the bytes that it read? Is the intention also to not check the length returned by `poll_read`?

Not that I have any idea which kind of bytes it would be returning.

Reply:
Yes. Quoting timerfd_create: `the buffer given to read(2) returns an unsigned 8-byte integer (uint64_t) containing the number of expirations that have occurred.  (The returned value is in host byte order—that is,  the  native byte order for integers on the host machine.)`
I don't see any useful application for this value for this usecase.